api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsArtCohortDataset.java

```diff
@@ -0,0 +1,238 @@
+
+  private TPTListOfPatientsEligibleDataSet tptListOfPatientsEligibleDataSet;
+
+  private TPTInitiationDataDefinitionQueries tptInitiationDataDefinitionQueries;
+
+  private ListChildrenOnARTandFormulationsDataset listChildrenOnARTandFormulationsDataset;
+
+  private ListOfPatientsEligibleForVLDataDefinitionQueries
+      listOfPatientsEligibleForVLDataDefinitionQueries;
+
+  @Autowired
+  public ListOfPatientsArtCohortDataset(
+      ListOfPatientsArtCohortCohortQueries listOfPatientsArtCohortCohortQueries,
+      TPTListOfPatientsEligibleDataSet tptListOfPatientsEligibleDataSet,
+      TPTInitiationDataDefinitionQueries tptInitiationDataDefinitionQueries,
+      ListChildrenOnARTandFormulationsDataset listChildrenOnARTandFormulationsDataset,
+      ListOfPatientsEligibleForVLDataDefinitionQueries
+          listOfPatientsEligibleForVLDataDefinitionQueries) {
+    this.listOfPatientsArtCohortCohortQueries = listOfPatientsArtCohortCohortQueries;
+    this.tptListOfPatientsEligibleDataSet = tptListOfPatientsEligibleDataSet;
+    this.tptInitiationDataDefinitionQueries = tptInitiationDataDefinitionQueries;
+    this.listChildrenOnARTandFormulationsDataset = listChildrenOnARTandFormulationsDataset;
+    this.listOfPatientsEligibleForVLDataDefinitionQueries =
+        listOfPatientsEligibleForVLDataDefinitionQueries;
+  }
+
+  public DataSetDefinition contructDataset() throws EvaluationException {
+
+    PatientDataSetDefinition pdd = new PatientDataSetDefinition();
+
+    pdd.setName("ARV");
+
+    PatientIdentifierType identifierType =
+        Context.getPatientService()
+            .getPatientIdentifierTypeByUuid("e2b966d0-1d5f-11e0-b929-000c29ad1d07");
+    DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+
+    DataDefinition identifierDef =
+        new ConvertedPatientDataDefinition(
+            "identifier",
+            new PatientIdentifierDataDefinition(identifierType.getName(), identifierType),
+            identifierFormatter);
+
+    DataConverter formatter = new ObjectFormatter("{familyName}, {givenName}");
+
+    DataDefinition nameDef =
+        new ConvertedPersonDataDefinition("name", new PreferredNameDataDefinition(), formatter);
+
+    pdd.setParameters(getParameters());
+
+    pdd.addRowFilter(
+        listOfPatientsArtCohortCohortQueries.getPatientsInitiatedART(),
+        "startDate=${startDate},endDate=${endDate},location=${location}");
+
+    /** 1- NID sheet 1 - Column A */
+    pdd.addColumn(
+        "nid",
+        tptListOfPatientsEligibleDataSet.getNID(identifierType.getPatientIdentifierTypeId()),
+        "");
+
+    /** 2 - Name - Sheet 1: Column B */
+    pdd.addColumn("name", nameDef, "");
+
+    /** 3 - Sexo - Sheet 1: Column C */
+    pdd.addColumn("gender", new GenderDataDefinition(), "", new GenderConverter());
+
+    /** 4 - Idade - Sheet 1: Column D */
+    pdd.addColumn("age", new AgeDataDefinition(), "", null);
```

Comment:
Any effective date that is based on, do we need to do as the reporting date or just today?